src/runner.js

```diff
@@ -107,7 +107,12 @@ export async function run(stdout, stdin, stderr, argv) {
   const errors = validateSchemaDefinition(schema, rules, configuration);
   const groupedErrors = groupErrorsBySchemaFilePath(errors, schema.sourceMap);
 
-  stdout.write(formatter(groupedErrors));
+  const writeResult = stdout.write(formatter(groupedErrors));
+  if (!writeResult) {
```

Comment:
I wasn't aware of this gotcha with `stdout.write`.

Thanks for the clear explanation.